### ISSUE TITLE
`scale_by_distance_over_gradients` returns `nan`

### ISSUE THREAD

**Issue by T90REAL** [google-deepmind/optax#1499]
The helper `_tx` inside `scale_by_distance_over_gradients` is defined as `_tx(g, d, g_sos)`, but `jax.tree.map` calls it with `(max_dist, g_sos, updates)`. As a result, `_tx` interprets the accumulated gradient sum-of-squares as `d` and the raw updates (which can be negative) as `g_sos`. When the raw gradient has any negative entry, the code executes `jnp.sqrt(g_sos + eps)` with a negative argument and produces NaN.

```python
import jax.numpy as jnp
import optax

params = [jnp.zeros(5)]
updates = [jnp.array([0., 0., 0., 0., -1.], dtype=jnp.float32)]

optimizer = optax.scale_by_distance_over_gradients(
    reps_rel=0.0078125,
    eps=6.103515625e-05,
    global_scale=1,
)

state = optimizer.init(params)
transformed_updates, _ = optimizer.update(updates, state, params)

print(transformed_updates[0])
```

```
[ 1.  1.  1.  1. nan]
```

**Comment by Aaryan-549** [comment]
Thanks for reporting this! I have opened a pull request to fix this issue here: https://github.com/google-deepmind/optax/pull/1501.

The error was indeed caused by an incorrect argument order when calling the internal _tx helper (passing max_dist where g was expected). The PR corrects the order to match the function signature.

**Comment by Aaryan-549** [comment]
@T90REAL Please close the issue as #1501 has been merged.
